pkg/sass-parser/lib/src/interpolation.ts

```diff
@@ -0,0 +1,422 @@
+// Copyright 2024 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import * as postcss from 'postcss';
+
+import {convertExpression} from './expression/convert';
+import {fromProps} from './expression/from-props';
+import {Expression, ExpressionProps} from './expression';
+import {LazySource} from './lazy-source';
+import {Node} from './node';
+import type * as sassInternal from './sass-internal';
+import * as utils from './utils';
+
+/**
+ * The type of new nodes that can be passed into an interpolation.
+ *
+ * @category Expression
+ */
+export type NewNodeForInterpolation =
+  | Interpolation
+  | Interpolation[]
+  | Expression
+  | Expression[]
```

Comment:
I suggest using `ReadonlyArray` for all those, to allow passing something typed as ReadonlyArray for the new node without having to copy the array first. The code in this class does not attempt to mutate the array anyway.

Reply:
Good call. I think we can actually make this change to a number of parameters, and potentially also upstream in the PostCSS types.